lib/services/awsAdapter.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+
+var Bluebird = require("bluebird");
+
+function AWSAdapter (ec2) {
+
+	var tags;
+	var instanceID;
+
+	this.runInstances = Bluebird.method(function (instance) {
+		var params = {
+			ImageId      : instance.ami,
+			InstanceType : instance.type,
+			MinCount     : 1,
+			MaxCount     : 1
+		};
+		return ec2.runInstancesAsync(params)
+		.then(function (data) {
+			instanceID = data.Instances[ 0 ].InstanceId;
+			tags = {
+				Resources : [ instanceID ],
+				Tags      : [ {
+					Key   : "Name",
+					Value : "Service Maker"
+				},
+				{
+					Key   : "ID",
+					Value : instance.id
+				} ]
+			};
+			ec2.createTagsAsync(tags);
+			return instance;
+		})
+		.catch(function (error) {
```

Comment:
This shouldn't be necessary-- if no `.catch` block exists, the error will continue to propogate to the next part of the chain (i.e., the function that is calling `runInstances`).